simple/template_schema.py

```diff
@@ -0,0 +1,159 @@
+    Use UCD controlled vocabulary: https://www.ivoa.net/documents/UCD1+/20200212/PEN-UCDlist-1.4-20200212.html#tth_sEcB
+    The string values are used, not the variable names.
+    """
+    ultraviolet = 'em.UV'
+    optical_UCD = 'em.opt'
+    optical = 'optical'
+    nir_UCD = 'em.IR.NIR'  # Near-Infrared, 1-5 microns
+    nir = 'nir'
+    infrared = 'em.IR'  # Infrared part of the spectrum
+    mir_UCD = 'em.IR.MIR'  # Medium-Infrared, 5-30 microns
+    mir = 'mir'
+    millimeter = 'em.mm'
+    radio = 'em.radio'
+    unknown = 'unknown'
+
+# -------------------------------------------------------------------------------------------------------------------
+# Main tables
+class Sources(Base):
+    """ORM for the sources table. This stores the main identifiers for our objects along with ra and dec"""
+    __tablename__ = 'Sources'
+    source = Column(String(100), primary_key=True, nullable=False)
+    ra = Column(Float)
+    dec = Column(Float)
+    epoch = Column(Float)  # decimal year
+    equinox = Column(String(10))  # eg, J2000
+    shortname = Column(String(30))  # not needed?
+    # reference = Column(String(30), ForeignKey('Publications.publication', onupdate='cascade'), nullable=False)
+    other_references = Column(String(100))
+    comments = Column(String(1000))
+
+
+class Names(Base):
+    __tablename__ = 'Names'
+    source = Column(String(100), ForeignKey('Sources.source', ondelete='cascade', onupdate='cascade'),
+                    nullable=False, primary_key=True)
+    other_name = Column(String(100), primary_key=True, nullable=False)
+
+# todo: make "tabulardata" or "physicaldata" abstract classes.
+
+
+class DataPointerTable(Base):
+    # don't want that table to be written! maybe it'll just get written anyways. but just an object.
+    # __tablename__ = 'DataPointerTable'
+    # don't take base here. try just have columns. multiple inheritance!
+    data_type = Column(String(32), nullable=False)
+    # Table to store references to data. like spectra, or light curves.
+    source = Column(String(100), ForeignKey('Sources.source', ondelete='cascade', onupdate='cascade'),
+                    nullable=False, primary_key=True)
+
+    # Data
+    spectrum = Column(String(1000), nullable=False)  # URL of spectrum location
+    original_spectrum = Column(String(1000)) # URL of original spectrum location, if applicable
+    local_spectrum = Column(String(1000))  # local directory (via environment variable) of spectrum location
+
+    # Common metadata
+    comments = Column(String(1000))
+    reference = Column(String(30), ForeignKey('Publications.publication', onupdate='cascade'), primary_key=True)
+    other_references = Column(String(100))
+    # __mapper_args__ = {'polymorphic_on': data_type}
+
+
+class Spectra(DataPointerTable):
+    # Table to store references to spectra
+    super().__init__()
+    __tablename__ = 'Spectra'
+    __mapper_args__ = {'polymorphic_identity': 'spectra'}
```

Comment:
I don't know what `polymorphic_identity` means; it may be a complexity you don't need to include in the template. If you DO need it, it may be worthwhile to add a link to a SQLAlchemy document that describes why.